src/Runner.Worker/Handlers/HandlerFactory.cs

```diff
@@ -67,7 +67,21 @@ public sealed class HandlerFactory : RunnerService, IHandlerFactory
                     var isLocalOptOut = StringUtil.ConvertToBoolean(Environment.GetEnvironmentVariable(Constants.Variables.Actions.AllowActionsUseUnsecureNodeVersion));
                     bool isOptOut = isWorkflowOptOutSet ? StringUtil.ConvertToBoolean(workflowOptOut) : isLocalOptOut;
                     if (!isOptOut)
-                    {
+                    {   
+                        var repoAction = action as Pipelines.RepositoryPathReference;
+                        var repoActionFullName = "";
+                        if (repoAction != null)
+                        {
+                            if (string.IsNullOrEmpty(repoAction.Name))
+                            {
+                                repoActionFullName = repoAction.Path; // local actions don't have a 'Name'
+                            }
+                            else
+                            {
+                                repoActionFullName = $"{repoAction.Name}/{repoAction.Path ?? string.Empty}".TrimEnd('/') + $"@{repoAction.Ref}";
+                            }
+                        }
+                        executionContext.Warning(string.Format(Constants.Runner.EnforcedNode12DetectedAfterEndOfLife, repoActionFullName));
```

Comment:
`repoActionFullName` may still be empty here.

The output will appear as:

```
The action  uses node12 which is deprecated and will be forced to run on node16. For more info: https://github.blog/changelog/2023-06-13-github-actions-all-actions-will-run-on-node16-instead-of-node12-by-default/"
```

Is that OK?

Reply:
Is that the case of running a local action like this?
```yaml
#...someyaml
uses: ./
```

Reply:
I decided to aggregate warnings from all steps to one warning. I attached screenshot with new warning and this example includes local action